src/interpreter/index.ts

```diff
@@ -1094,34 +1091,10 @@ export class Interpreter implements Expr.Visitor<BrsType>, Stmt.Visitor<BrsType>
 
                     return arg;
                 });
-
-                if (
-                    expression.callee instanceof Expr.DottedGet ||
-                    expression.callee instanceof Expr.IndexedGet
-                ) {
-                    if (expression.callee.obj instanceof Expr.Call) {
-                        mPointer = this._lastDotGetAA;
-                    } else {
-                        let maybeM = this.evaluate(expression.callee.obj);
-                        maybeM = isBoxable(maybeM) ? maybeM.box() : maybeM;
-
-                        if (maybeM.kind === ValueKind.Object) {
-                            if (maybeM instanceof RoAssociativeArray) {
-                                mPointer = maybeM;
-                            }
-                        } else {
-                            return this.addError(
-                                new BrsError(
-                                    "Attempted to retrieve a function from a primitive value",
-                                    expression.closingParen.location
-                                )
-                            );
-                        }
-                    }
-                } else {
-                    this._lastDotGetAA = this.environment.getRootM();
+                let mPointer = this._environment.getRootM();
+                if (expression.callee instanceof Expr.DottedGet) {
```

Comment:
What happens with code like this?
```
function test()
  m.getMovie = function()
    return {
      movieId: 123
      getMovieId: function()
        return m.movieId
      end function
    }
  end function
  print m.getMovie().getMovieId()
end function

Reply:
It works, it prints `123` like expected.
The evaluation that existed on this method was redundant.

Reply:
Sounds good!